sdk/modelsrepository/Azure.Iot.ModelsRepository/src/Fetchers/LocalModelFetcher.cs

```diff
@@ -65,7 +65,7 @@ public FetchResult Fetch(string dtmi, Uri repositoryUri, CancellationToken cance
                     fnfError = string.Format(CultureInfo.CurrentCulture, ServiceStrings.ErrorFetchingModelContent, tryContentPath);
                 }
 
-                throw new RequestFailedException(fnfError, new FileNotFoundException(fnfError));
```

Comment:
Based on `RequestFailedException` documentation, it only applies to requests that have an HTTP status. 
This exception is caught at `RepositoryHandler` level and converted to a `ResolverException` which is currently the Exception that is exposed to users.
We will have a discussion about the Exception that is thrown from the ResolverException point of view during the initial API review

Reply:
Let's add doc comments to the client that indicate what exception types can be thrown and when.